### ISSUE TITLE
Mythic Beasts provider does not use documented config params for DNS propagation timeout and polling interval

### ISSUE THREAD

**Issue by codebrewer** [go-acme/lego#1357]
The documentation for the Mythic Beasts provider lists the following additional configuration variables:

| Environment Variable Name | Description |
|--------------------------------|-------------|
| `MYTHICBEASTS_POLLING_INTERVAL` | Time between DNS propagation check |
| `MYTHICBEASTS_PROPAGATION_TIMEOUT` | Maximum waiting time for DNS propagation |

but when using the propagation timeout variable:

```bash
docker inspect traefik | jq '.[0].Config.Env'
[
  "MYTHICBEASTS_PASSWORD=***MASKED***",
  "MYTHICBEASTS_PROPAGATION_TIMEOUT=120000000000",
  "MYTHICBEASTS_USERNAME=***MASKED***",
  "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
]
```

I see the default value being used:

```
time="2021-02-27T12:57:30Z" level=debug msg="legolog: [INFO] Wait for propagation [timeout: 1m0s, interval: 2s]"
```

Looking at `providers/dns/mythicbeasts/mythicbeasts.go` suggests this happens because the Mythic Beasts provider doesn't implement the `ProviderTimeout` interface.

I'm happy to try fixing and raise a PR - I'm a Go newbie but there are plenty of other implementations to look at. Some guidance on an appropriate test would be useful though (although I don't see any in the few provider-specific test files I've looked at).

**Comment by ldez** [comment]
You just have to add:

```go

// Timeout returns the timeout and interval to use when checking for DNS propagation.
// Adjusting here to cope with spikes in propagation times.
func (d *DNSProvider) Timeout() (timeout, interval time.Duration) {
	return d.config.PropagationTimeout, d.config.PollingInterval
}
```

**Comment by codebrewer** [comment]
Yeah, thanks, I have it working with that very change. I discovered I got the units wrong for the values though...

What about a test? Or will you accept a PR with just that change?

**Comment by palvarez89** [comment]
Thank you both for addressing this issue!

@ldez Any chance this can go into a release soon-ish?

**Comment by palvarez89** [comment]
> Thank you both for addressing this issue!
> 
> Any chance this can go into a release soon-ish?

Included in [v4.3.0](https://github.com/go-acme/lego/commits/v4.3.0) THANKS!!!
